### PR TITLE
Clamp root spill size

### PR DESCRIPTION
**Pull request by beckerg**
- Add rparam rspill_sizemb_max to limit max rspill size.
- Add comments to guide future zero-writeamp root spill operations.

### CODE REVIEW

**Comment by beckerg** [comment]
This patch evens out root spill by size which reduces the amount of boxcarring that happens when short-rspill threads end up having to wait on long-rspill threads (which deprives other kvs root from being serviced in a timely manner).
The refactoring sets us up to detect situations in which we can issue a zero-writeamp spill vs the kv-compaction based spill that we currently always do today.

File: lib/cn/csched_sp3_work.c [hunk 1]
```diff
@@ -145,49 +154,56 @@ sp3_work_rspill_find_kvsets(struct sp3_node *spn, uint n_max, struct kvset_list_
...
     while ((le = list_prev_entry_or_null(le, le_link, &tn->tn_kvset_list))) {
-        if (n_kvsets == n_max)
-            break;
-
         if (kvset_get_workid(le->le_kvset) != 0)
             break;
 
-        n_kvsets++;
-    }
+        wlen += kvset_get_kwlen(le->le_kvset) + kvset_get_vwlen(le->le_kvset);
```
Comment (alexttx) [inline lib/cn/csched_sp3_work.c:173]: add wlen twice?

File: lib/cn/csched_sp3_work.h [hunk 2]
```diff
@@ -29,24 +29,29 @@ enum sp3_work_type {
...
  * node length reduction requires at least 2 kvsets.
  */
-#define SP3_RSPILL_KVSETS_MIN   ((u8)1)
-#define SP3_LLEN_RUNLEN_MIN     ((u8)2)
+#define SP3_RSPILL_RUNLEN_MIN   (1u)
+#define SP3_RSPILL_RUNLEN_MAX   (16u)
+#define SP3_RSPILL_SIZEMB_MIN   (4u * 1024)
+#define SP3_RSPILL_SIZEMB_MAX   (32u * 1024)
+#define SP3_LLEN_RUNLEN_MIN     (2u)
+#define SP3_LLEN_RUNLEN_MAX     (16u)
```
Comment (alexttx) [inline lib/cn/csched_sp3_work.h:54]: maybe add defs for default values and replace the literals that are using when the thresholds are initialized.

**Review by alexttx — APPROVED** [review]

File: lib/cn/csched_sp3_work.h (hunk 2 above)
Comment (beckerg) [reply lib/cn/csched_sp3_work.h:54]: Was thinking the same, can do next time I update these when we use more metrics for other job triggers...

**Review by tristan957 — APPROVED** [review]

**Pull request merged**